src/twisted/internet/defer.py

```diff
@@ -920,7 +926,7 @@ def errback(self, fail: Optional[Union[Failure, BaseException]] = None) -> None:
         """
         if fail is None:
             fail = Failure(captureVars=self.debug)
-        elif not isinstance(fail, Failure):
+        elif not hasattr(fail, "_twistedPrivateIsFailure"):
```

Comment:
I'm less worried about the `hasattr()` in this case, Exceptions tend to do much. Still a potential issue, but probably unlikely enough so as not to matter.